### PR TITLE
Add action and note services, mutations, querys and tests

### LINKED ISSUE #195: Backend: Action and notes mutation

### PR DESCRIPTION
**Pull request by Christdej**
Closes #195

### CODE REVIEW

File: backend/api/GQL/Mutation.cs [hunk 1]
```diff
@@ -88,25 +96,64 @@ public Participant DeleteParticipant(string participantId)
...
+            return _noteService.Create(CreatedBy(evaluation), text, action);
+        }
+
+        public Note EditNote(string noteId, string text)
+        {
+            Note note = _noteService.GetNote(noteId);
+            return _noteService.EditNote(note, text);
+        }
+
+        private Participant CreatedBy(Evaluation evaluation)
```
Comment (nicholasdalhaug) [inline backend/api/GQL/Mutation.cs:153]: Confusing naming?

```suggestion
        private Participant GetCurrentParticipant(Evaluation evaluation)
```

File: backend/api/GQL/Mutation.cs [hunk 2]
```diff
@@ -88,25 +96,64 @@ public Participant DeleteParticipant(string participantId)
...
         public Answer SetAnswer(string questionId, Severity severity, string text, Progression progression)
         {
-            string azureUniqueId = _authService.GetOID();
-
             IQueryable<Question> queryableQuestion = _questionService.GetQuestion(questionId);
             Question question = queryableQuestion.First();
             Evaluation evaluation = queryableQuestion.Select(q => q.Evaluation).First();
-            Participant participant = _participantService.GetParticipant(azureUniqueId, evaluation);
-
+            Participant createdBy = CreatedBy(evaluation);
```
Comment (nicholasdalhaug) [inline backend/api/GQL/Mutation.cs:102]:
```suggestion
            Participant currentParticipant = CreatedBy(evaluation);
```

File: backend/api/GQL/Mutation.cs [hunk 3]
```diff
@@ -88,25 +96,64 @@ public Participant DeleteParticipant(string participantId)
...
             catch (NotFoundInDBException)
             {
-                answer = _answerService.Create(participant, question, severity, text, progression);
+                answer = _answerService.Create(createdBy, question, severity, text, progression);
             }
 
             return answer;
         }
+
+        public Action CreateAction(string questionId, string assignedToId, string description, DateTime dueDate, string title)
```
Comment (nicholasdalhaug) [inline backend/api/GQL/Mutation.cs:117]: Missing `priority` field.

File: backend/api/Services/ActionService.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+            Action newAction = new Action
+            {
+                CreateDate = createDate,
+                AssignedTo = assignedTo,
+                CreatedBy = createdBy,
+                Description = description,
+                DueDate = dueDate,
+                OnHold = false,
+                Priority = Priority.Low,
```
Comment (nicholasdalhaug) [inline backend/api/Services/ActionService.cs:38]: Shouldn't `priority` be a parameter?

File: backend/api/Services/ActionService.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                Title = title,
+                Question = question
```
Comment (nicholasdalhaug) [inline backend/api/Services/ActionService.cs:40]: Should there be some sort of `isCompleted` field?

File: backend/tests/Services/ActionService.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            Action actionGet = actionService.GetAction(actionCreate.Id).First();
+
+            Assert.Equal(actionCreate, actionGet);
+        }
+
+        [Fact]
+        public void EditAction()
+        {
+            ParticipantService participantService = new ParticipantService(_context);
+            Participant participant = participantService.GetAll().ToList()[2];
```
Comment (nicholasdalhaug) [inline backend/tests/Services/ActionService.cs:69]: Why not just use `First()` here?

File: backend/tests/Services/NoteService.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+            IQueryable<Note> noteQueryable = noteService.GetAll();
+
+            Assert.True(noteQueryable.Count() > 0);
+        }
+
+        [Fact]
+        public void Create()
+        {
+            ParticipantService participantService = new ParticipantService(_context);
+            Participant participant = participantService.GetAll().ToList()[0];
```
Comment (nicholasdalhaug) [inline backend/tests/Services/NoteService.cs:29]: Why not just use `First()` here?

**Review by nicholasdalhaug — CHANGES_REQUESTED** [review]

File: backend/api/Services/ActionService.cs (hunk 5 above)
Comment (Christdej) [reply backend/api/Services/ActionService.cs:40]: Im waiting for a replay in the teams channel

File: backend/tests/Services/NoteService.cs (hunk 7 above)
Comment (Christdej) [reply backend/tests/Services/NoteService.cs:29]: No good excuse xD will fix

File: backend/tests/Services/ActionService.cs (hunk 6 above)
Comment (Christdej) [reply backend/tests/Services/ActionService.cs:69]: No good excuse xD will fix

File: backend/api/GQL/Mutation.cs (hunk 1 above)
Comment (Christdej) [reply backend/api/GQL/Mutation.cs:153]: Yes, totally agreed

File: backend/api/GQL/Mutation.cs (hunk 3 above)
Comment (Christdej) [reply backend/api/GQL/Mutation.cs:117]: Will fix

File: backend/api/Services/ActionService.cs (hunk 4 above)
Comment (Christdej) [reply backend/api/Services/ActionService.cs]: It should!

File: backend/api/GQL/Mutation.cs (hunk 2 above)
Comment (Christdej) [reply backend/api/GQL/Mutation.cs]: agreed

**Review by nicholasdalhaug — APPROVED** [review]

**Pull request merged**